### DISCUSSION TITLE
What exactly do you need Microsoft WebView in Orca Slicer for ?

### DISCUSSION THREAD

**Discussion by grumpy-a-f** [OrcaSlicer/OrcaSlicer#4345] [General]
when you start Orca Slicer under Windows - it spawns a bunch of threads 'msedgewebview2.exe'. what one would need that for. please ?

if access to that Microsoft folder is disabled - on slicer start up you get the message: 

"Orca Slicer requires the Microsoft WebView2 Runtime to operate certain features. Click Yes to install it now"

anyone care to explain what are those 'certain features', and why I would need them ?

**Comment by TexZeTech** [comment]
I believe it would have to do with the device tab. However this is just my assumption.

**Reply by AdamDrefs** [reply]
I also believe this to be the case, since when there is no connection, it shows the same page as Microsoft Edge, and the right-click menu is the same as in Edge.

**Comment by LJKraus** [comment]
I believe it is used to provide the video.

**Comment by zombodotcom** [comment]
used for the setup wizard inside orcaslicer

**Comment by gregazar** [comment]
Actually, webview is what loads the web based content such as MakerWorld Model Content and Project Information (even though the project info is stored in the 3mf)

I completely removed Edge and all related components from my machines and blocked them from returning. I also get that annoying prompt to install webview every time I launch Orca. Now, BB Studio will not prompt you to install webview unless you are logged into your BBL account.

<img width="1919" height="1079" alt="2026-02-22_21-00-31" src="https://github.com/user-attachments/assets/8ef97ee0-6ca8-40ee-a093-1c69f17a1afa" />
<img width="1919" height="1079" alt="2026-02-22_20-57-44" src="https://github.com/user-attachments/assets/da8e71b0-2d3a-4f3a-9b21-f29e130ba468" />

**Comment by yw4z** [comment]
Here is a list for parts that uses web dialogs
• Setup Wizard
• Filament Selection Dialog
• Printer Selection Dialog
• Flushing Volumes Dialog
• Home page
• Device Page (non bambu ones, klipper etc.)
• Project page
• BBL login from homepage
• Release Notes dialog partly uses a html frame

Probably these list will increase overtime because developing some dialogs with web interface much more easier sometimes
we saw a switch for Flushing Volumes Dialog to web UI recently

macOS and Linux uses their self renderers for web content

Problem with windows, they didnt integrated properly with their OS. still it works like a addon to system instead full integration. macOS and Linux uses a proper integration with OS
